schedule/__init__.py

```diff
@@ -75,9 +75,17 @@ def run_all(self, delay_seconds=0):
             self._run_job(job)
             time.sleep(delay_seconds)
 
-    def clear(self):
-        """Deletes all scheduled jobs."""
-        del self.jobs[:]
+    def clear(self, tag=None):
+        """Deletes scheduled jobs marked with the given tag, or all jobs
+        if tag is omitted"""
+        if tag is None:
+            del self.jobs[:]
+        else:
+            old_ref = self.jobs
+            self.jobs = [job for job in self.jobs if tag not in job.tags]
+            global jobs
+            if jobs is old_ref:
+                jobs = self.jobs
```

Comment:
Here's something to consider: I just realized this changes the current behavior – previously `schedule.jobs` (the module-level global) was only ever modified and not deleted/recreated. So if someone holds a reference to that list and then deletes a tag they're not going to see the updates.

I don't really see how we could modify the jobs list in place in a sensible way though (it's going to be slow). Any thoughts on this?